aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1316,51 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
+    try {
+      Class<?> providerClass =
+          DynClasses.builder()
+              .loader(this.getClass().getClassLoader())
```

Comment:
Any reason why we are specifying the classloader here?

Reply:
Removed ClassLoader. By default DynClasses api uses context class loader